src/test/stepFunctions/commands/visualizeStateMachineCDK.test.ts

```diff
@@ -0,0 +1,411 @@
+    vscode.TreeItemCollapsibleState.Collapsed,
+    mockStateMachineConstructTreeEntity
+)
+
+const mockStateMachineNodeSameAppDiffName = new ConstructNode(
+    new FakeParentNode('cdkJsonPath'),
+    'MyCDKApp1/MyStateMachine2',
+    vscode.TreeItemCollapsibleState.Collapsed,
+    mockStateMachineConstructTreeEntity
+)
+
+describe('StepFunctions VisualizeStateMachine', async function () {
+    const oldWebviewScriptsPath = ext.visualizationResourcePaths.localWebviewScriptsPath
+    const oldWebviewBodyPath = ext.visualizationResourcePaths.webviewBodyScript
+    const oldCachePath = ext.visualizationResourcePaths.visualizationLibraryCachePath
+    const oldScriptPath = ext.visualizationResourcePaths.visualizationLibraryScript
+    const oldCssPath = ext.visualizationResourcePaths.visualizationLibraryCSS
+    const oldThemePath = ext.visualizationResourcePaths.stateMachineCustomThemePath
+    const oldThemeCssPath = ext.visualizationResourcePaths.stateMachineCustomThemeCSS
+
+    // Before all
+    before(function () {
+        ext.visualizationResourcePaths.localWebviewScriptsPath = mockUri
+        ext.visualizationResourcePaths.visualizationLibraryCachePath = mockUri
+        ext.visualizationResourcePaths.stateMachineCustomThemePath = mockUri
+        ext.visualizationResourcePaths.webviewBodyScript = mockUri
+        ext.visualizationResourcePaths.visualizationLibraryScript = mockUri
+        ext.visualizationResourcePaths.visualizationLibraryCSS = mockUri
+        ext.visualizationResourcePaths.stateMachineCustomThemeCSS = mockUri
+
+        sandbox = sinon.createSandbox()
+        sandbox.stub(StateMachineGraphCache.prototype, 'updateCachedFile').callsFake(async options => {
+            return
+        })
+    })
+
+    // Before each
+    beforeEach(function () {
+        mockAslVisualizationCDKManager = new MockAslVisualizationCDKManager(mockExtensionContext, 'Workspace1')
+    })
+
+    // After all
+    after(function () {
+        sandbox.restore()
+        ext.visualizationResourcePaths.localWebviewScriptsPath = oldWebviewScriptsPath
+        ext.visualizationResourcePaths.webviewBodyScript = oldWebviewBodyPath
+        ext.visualizationResourcePaths.visualizationLibraryCachePath = oldCachePath
+        ext.visualizationResourcePaths.visualizationLibraryScript = oldScriptPath
+        ext.visualizationResourcePaths.visualizationLibraryCSS = oldCssPath
+        ext.visualizationResourcePaths.stateMachineCustomThemePath = oldThemePath
+        ext.visualizationResourcePaths.stateMachineCustomThemeCSS = oldThemeCssPath
+    })
+
+    // Tests
+    it('Test AslVisualizationCDK on setup all properties are correct', function () {
+        const vis = new MockAslVisualizationCDK(mockTextDocument, '', '', '')
+
+        assert.deepStrictEqual(vis.documentUri, mockTextDocument.uri)
+        assert.strictEqual(vis.getIsPanelDisposed(), false)
+        assert.strictEqual(vis.getDisposables().length, 5)
+
+        let panel = vis.getPanel()
```

Comment:
Make const and assert `as vscode.WebviewPanel` at this level. Same with `as vscode.Webview` on like 192.

Reply:
Unsure why this was resolved. This should read `const panel = vis.getPanel() as vscode.WebviewPanel`; there's no reason to reassign it to itself later.